lib/compress/zstd_compress_internal.h

```diff
@@ -148,6 +149,7 @@ typedef struct {
     U32 bucketSizeLog;      /* Log bucket size for collision resolution, at most 8 */
     U32 minMatchLength;     /* Minimum match length */
     U32 hashEveryLog;       /* Log number of entries to skip */
+    U32 windowLog;          /* Window log for the LDM */
 } ldmParams_t;
```

Comment:
I'm slightly embarrassed here by the discrepancy between the name of the structure, and its actual function.

While `minMatchLength` and such do look like `params`,
`hashTable`, `bucketOffsets` and now `window` feel much more like a state context, that must be transported and evolves during compression.

It sounds like the 2 functions are mixed.

What I would expect in such a case is for such a structure to be actually named a `state` (or a `context`), _inside which_ one can find a nested `param` structure, which content is limited to just params (and might be used independently for other purpose, without all the `state` values attached).

What do you think ?